core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -3138,24 +3143,14 @@ protected void validateFrom(
     case UNNEST:
       validateUnnest((SqlCall) node, scope, targetRowType);
       break;
-    case LATERAL:
-      // Validate subquery that LATERAL precedes
-      validateQuery(((SqlCall) node).operand(0), scope, targetRowType);
-      break;
```

Comment:
This was incorrect, we want to recurse into same function (`validateFrom`) and not into `validateQuery`.